File: app/src/main/java/fragments/BankersFragment.java
package fragments;
/*
    This fragment is attached to Profile activity
 */

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.Query;
import com.sqube.tipshub.R;

import adapters.BankerAdapter;
import utils.Calculations;
import utils.FirebaseUtil;

public class BankersFragment extends Fragment {
    private String userId, myId;
    private BankerAdapter postAdapter;
    private RecyclerView recyclerView;
    private FirebaseUser user;

    public BankersFragment() {
        // Required empty public constructor
    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView=inflater.inflate(R.layout.fragment_bankers, container, false);
        recyclerView = rootView.findViewById(R.id.postList);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        user = FirebaseUtil.getFirebaseAuthentication().getCurrentUser();
        if(user==null)
            myId= Calculations.GUEST;
        else
            myId = user.getUid();
        userId = getArguments().getString("userId");
        loadPost();
        return rootView;
    }

    private void loadPost() {
        String TAG = "PostFragment";
        Log.i(TAG, "loadPost: ");
        Query query = FirebaseUtil.getFirebaseFirestore().collection("posts").orderBy("time", Query.Direction.DESCENDING)
                .whereEqualTo("userId", userId).whereEqualTo("type", 6);
        postAdapter = new BankerAdapter(query, myId, getContext());
        recyclerView.setAdapter(postAdapter);
        if(postAdapter!=null){
            Log.i(TAG, "loadPost: started listening");
            postAdapter.startListening();
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        user = FirebaseUtil.getFirebaseAuthentication().getCurrentUser();
        if(user==null)
            myId= Calculations.GUEST;
        else
            myId = user.getUid();
        if(postAdapter!=null)
            postAdapter.setUserId(myId);
    }

}

File: app/src/main/java/fragments/BankersFragment.kt
package fragments

import adapters.BankerAdapter
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.firestore.Query
import com.sqube.tipshub.R
import utils.Calculations
import utils.FirebaseUtil

/*
    This fragment is attached to Profile activity
 */   class BankersFragment : Fragment() {
    private var userId: String? = null
    private var myId: String? = null
    private var postAdapter: BankerAdapter? = null
    private var recyclerView: RecyclerView? = null
    private var user: FirebaseUser? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val rootView = inflater.inflate(R.layout.fragment_bankers, container, false)
        recyclerView = rootView.findViewById(R.id.postList)
        recyclerView?.setLayoutManager(LinearLayoutManager(activity))
        user = FirebaseUtil.firebaseAuthentication?.currentUser
        myId = if (user == null) Calculations.GUEST else user!!.uid
        userId = requireArguments().getString("userId")
        loadPost()
        return rootView
    }

    private fun loadPost() {
        val TAG = "PostFragment"
        Log.i(TAG, "loadPost: ")
        val query = FirebaseUtil.firebaseFirestore?.collection("posts")?.orderBy("time", Query.Direction.DESCENDING)!!
                .whereEqualTo("userId", userId).whereEqualTo("type", 6)
        postAdapter = BankerAdapter(query, myId, context, false)
        recyclerView!!.adapter = postAdapter
        if (postAdapter != null) {
            Log.i(TAG, "loadPost: started listening")
            postAdapter!!.startListening()
        }
    }

    override fun onResume() {
        super.onResume()
        user = FirebaseUtil.firebaseAuthentication?.currentUser
        myId = if (user == null) Calculations.GUEST else user!!.uid
        if (postAdapter != null) postAdapter!!.setUserId(myId)
    }
}